Change build system from Ant to Gradle

egor-bukharov opened the pull request:
Add integration test for screenshotTaker class

egor-bukharov committed:
Change build system from Ant to Gradle
Add integration test for screenshotTaker class

AlexanderMakarov commented on build.gradle:
```diff
@@ -0,0 +1,70 @@
+
+        def adb = "$androidHome/platform-tools/adb"
+
+        javaexec {
```
Windows Server 2016 R2, Gradle 2.2.1 (yes, not wrapper).
```
d:\tmp\screemshotTaker\screenshotTaker>gradle integrationTest -PdeviceSerial=emulator-5554
:compileJava
:processResources UP-TO-DATE
:classes
:compileIntTestJava
:processIntTestResources UP-TO-DATE
:intTestClasses
:integrationTest
Error: Unable to access jarfile D:\tmp\screemshotTaker\screenshotTaker/screenshotTaker.jar
:integrationTest FAILED

FAILURE: Build failed with an exception.

* Where:
Build file 'D:\tmp\screemshotTaker\screenshotTaker\build.gradle' line: 49

* What went wrong:
Execution failed for task ':integrationTest'.
> Process 'command 'C:\Program Files\Java\jdk1.8.0_77\bin\java.exe'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 26.204 secs
```

AlexanderMakarov commented on build.gradle:
```diff
@@ -0,0 +1,70 @@
+
+    doFirst {
+        if (!project.hasProperty('deviceSerial')) {
+            throw new GradleException('deviceSerial property value must be provided for the task. Usage: gradle integrationTest -PdeviceSerial=<your-device-serial>')
```
>Usage: gradle integrationTest -PdeviceSerial=<your-device-serial>'

Why it suggest to use local gradle executor (with unknown version) instead of built-in gradle wrapper (`gradlew`, with unknown version)?

AlexanderMakarov commented on gradlew.bat:
```diff
@@ -0,0 +1,84 @@
+@if "%DEBUG%" == "" @echo off
```
Windows Server R2. Seems like `$APP_HOME/gradle/wrapper/gradle-wrapper.jar` is missed.

```
d:\tmp\screemshotTaker\screenshotTaker>gradlew.bat integrationTest -PdeviceSerial=emulator-5554
Error: Could not find or load main class org.gradle.wrapper.GradleWrapperMain

d:\tmp\screemshotTaker\screenshotTaker>gradlew.bat help
Error: Could not find or load main class org.gradle.wrapper.GradleWrapperMain                
```

AlexanderMakarov commented on settings.gradle:
```diff
@@ -0,0 +1 @@
+rootProject.name = 'screenshotTaker'
```
User can clone repo into folder with other name. [Gradle itself knows path to project folder](https://stackoverflow.com/questions/41635131/gradle-getting-the-root-project-directory-path-when-starting-with-a-custom-buil/41635558).

AlexanderMakarov commented on src/intTest/java/src/ScreenshotTakerTest.java:
```diff
@@ -0,0 +1,70 @@
+import java.io.IOException;
+
+import static junit.framework.TestCase.assertTrue;
+import static junit.framework.TestCase.fail;
```
>import org.junit.Test;

Is not used.

>import static junit.framework.TestCase.assertTrue;
>import static junit.framework.TestCase.fail;

From whole JUnit you are using only methods to make unchecked exceptions. `assertTrue` is equal to `if (condition) throw AssertException(some logic to build clear message)`.

1. They can be implemented manually, not need to to ask JUnit dependency only for that.
1. You are interested only in process exit code, such exceptions can be replaced with printing error details and `System.exit(1)`.

AlexanderMakarov commented on build.gradle:
```diff
@@ -0,0 +1,70 @@
+
+task integrationTest(type: JavaExec) {
+    def imageFromAppFileName = 'screenshotTakerTest.png'
+    def imageFromAdbFileName = 'adbScreenshotTest.png'
```
It is better to add description for task to get something in `gradle tasks` output.

Now it looks like:
```
Other tasks
-----------
integrationTest

Rules                
```

egor-bukharov committed:
Update README.md
Replace JUnit assertions with exceptions
Add gradle-wrapper.jar
Add description for `integrationTest` task

egor-bukharov committed:
Delete redundant settings.gradle file

egor-bukharov committed:
Remove unnecessary `file()` from build.gradle
